packages/mobile-native-mcp-server/src/workflow/magi/prd/nodes/prdMagiInitialization.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { PRDState } from '../metadata.js';
+import { PRDAbstractToolNode } from './prdAbstractToolNode.js';
+import { ToolExecutor } from '../../../nodes/toolExecutor.js';
+import { Logger } from '../../../../logging/logger.js';
+import { ensureMagiSddDirectory } from '../../../../utils/wellKnownDirectory.js';
+
+export class PRDMagiInitializationNode extends PRDAbstractToolNode {
+  constructor(toolExecutor?: ToolExecutor, logger?: Logger) {
+    super('magiInitialization', toolExecutor, logger);
+  }
+
+  execute = (state: PRDState): Partial<PRDState> => {
+    // Initialize state with user input and default values
+    // Only set defaults for fields that don't already exist (to avoid overwriting existing state)
+    const initializedState: Partial<PRDState> = {
```

Comment:
It feels like maybe this whole block should be extracted out to a reusable service. I see a number of places in our nodes where we have some subset of pre-initializing state. We should just have a function that does all of it, that other nodes can call.